### PR TITLE
fix `form.elements.length has incorrect value if a form has a file input` (close #2009)

### LINKED ISSUE #2009: form.elements.length has incorrect value if a form has a file input
We append a hidden input element to forms that upload files. All form's inputs can be accessed via the Array-like `form.elements` property. When you try to access our hidden input via `form.elements[i]` it's filtered and `undefined` is retured instead. But it still affects the `form.elements.length` property - it increases it by 1. 

##### test.html
```html
<html>
    <head>
        <title>TEST</title>
    </head>
    <html>
        <form id="form">
            <input type="file">
        </form>
        <button id="button">Click me</button>
        <div id="div"></div>
        <script>
            const el = form.elements[1];

            button.onclick = () => div.textContent = `form.elements.length: ${form.elements.length} form.elements[1]: ${typeof el}`;
        </script>
    </html>
</html>
```

##### test.js
```js
import { Selector } from "testcafe";

fixture`form.elements.length`.page('./test.html');

test(`test`, async t => {
    await t
        .setFilesToUpload('input', 'test.js')
        .click('#button')
        .expect(Selector('#div').textContent).eql(`form.elements.length: 1 form.elements[1]: undefined`);
});
```

### PR DESCRIPTION
**Pull request by Farfurix**
https://github.com/DevExpress/testcafe-hammerhead/issues/2009
https://github.com/DevExpress/testcafe-hammerhead/issues/2010

### Changes
1. Form with a hidden input: mark `form.elements`, `form.children` and `form.childNodes` as shadow collections.
2. Form with a hidden input: mark the form as a shadow container (the `childElementCount` case).

### CODE REVIEW

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

File: src/client/sandbox/upload/hidden-info.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@ function createInput (form) {
     nativeMethods.inputValueSetter.call(hiddenInput, '[]');
     nativeMethods.appendChild.call(form, hiddenInput);
 
+    ShadowUI.markFormAsShadow(form);
```
Comment (miherlosev) [inline src/client/sandbox/upload/hidden-info.ts]: I guess we should leave `hiddenInput`  as the last child item of the `form`. It allows us to exclude its from loops (`for`,`for ...of`).

File: test/client/fixtures/sandbox/upload-test.js [hunk 2]
```diff
@@ -159,6 +159,24 @@ function getFiles (filesInfo) {
...
+    var form  = document.createElement('form');
+    var input = document.createElement('input');
+
+    document.body.appendChild(form);
+    form.appendChild(input);
+
+    return uploadSandbox.doUpload(input, './file.txt')
+        .then(function () {
+            strictEqual(form.elements.length, 1);
+            strictEqual(form.children.length, 1);
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/upload-test.js]: Add tests for loops.

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

**Pull request merged**